src/Validation/Validator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation;
+
+use Lcobucci\JWT\Token;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ *
+ * @since 4.0.0
+ */
+final class Validator implements \Lcobucci\JWT\Validator
```

Comment:
@henriquemoody what I dislike about having a result object is the requiring users to do this when they just care about the result:

```php
if (!$validator->validate($token, ...$constraints)->isValid()) { // or assign a variable for the result ofc
}
```

> Split it into two methods may be a good idea but also looks too much responsibility for the `Validator`.

Both methods are related to the same thing, so the reason to change is the same.

@dannydorfel @henriquemoody @schnittstabil @Ocramius thanks for your suggestions regarding this. I think the best option for now is splitting in two methods.